src/MyArrayList.java

```diff
@@ -0,0 +1,299 @@
+import java.util.Arrays;
+import java.util.Iterator;
+
+public class MyArrayList implements MyList, Iterable {
+
+    private int count;
+    private Object[] array;
+
+    private MyArrayList() {
+        count = 0;
+        array = new Object[10];
+    }
+
+    private void printAll() {
+        System.out.println(Arrays.toString(array));
+    }
+
+    private void expand(int index) {
+        int temp = this.array.length;
+
+        while (index >= temp) {
+            temp = temp * 2;
+        }
+
+        Object[] array = new Object[temp];
+        System.arraycopy(this.array, 0, array, 0, this.array.length);
+        this.array = array;
+    }
+
+    private void shiftRight(int index) {
+        Object[] array = new Object[this.array.length];
+        System.arraycopy(this.array, 0, array, 0, index);
+        System.arraycopy(this.array, index, array, index + 1, this.array.length - index - 1);
+        this.array = array;
+    }
+
+    private void shiftLeft(int index) {
+        Object[] array = new Object[this.array.length];
+        System.arraycopy(this.array, 0, array, 0, this.array.length);
+        System.arraycopy(array, index + 1, this.array, index, array.length - index - 1);
+    }
+
+    @Override
+    public void add(Object elem) {
+        add(count, elem);
+    }
+
+    @Override
+    public void add(int index, Object elem) {
+        if (index < count) {
+            shiftRight(index);
+            array[index] = elem;
+        }
+
+        else if (index < array.length) {
+            array[index] = elem;
+        }
+
+        else {
+            expand(index);
+            array[index] = elem;
+        }
+
+        count = index > count ? ++index : ++count;
+    }
+
+    @Override
+    public Object set(int index, Object elem) {
+
+        Object temp;
+
+        if (index >= array.length) {
+              expand(index);
+              array[index] = elem;
+              temp = null;            //здесь мог быть Exception
+          }
+
+         else {
+              temp = array[index];
+              array[index] = elem;
+          }
+
+        count = index > count ? ++index : count;
+
+        return temp;
+    }
+
+    @Override
+    public boolean remove(Object elem) {
+        int temp = indexOf(elem);
+
+        if (temp == -1) {
+            return false;
+        }
+
+        else {
+            remove(temp);
+            return true;
+        }
+
+    }
+
+    @Override
+    public Object remove(int index) {
+        Object temp = this.array[index];        //здесь может быть Exception
+        shiftLeft(index);
+        count--;
+
+        return temp;
+    }
+
+    @Override
+    public void clear() {
+        Object[] array = new Object[this.array.length];
+        Arrays.fill(array, 0, this.array.length, null);
+        this.array = array;
+        count = 0;
+    }
+
+    @Override
+    public Object get(int index) {
+        return array[index];            //здесь может быть Exception
```

Comment:
Do the following:
if index is out of range then throw IndexOutOfBoundsException
Why does that needs to be done by hand? Because if not then it will throw _Array_IndexOutOfBoundsException which is not ideologically correct because the persone who is going to use your list will operate not an array but list.